subsys/net/lib/sockets/sockets.c

```diff
@@ -684,7 +684,11 @@ int z_impl_zsock_accept(int sock, struct sockaddr *addr, socklen_t *addrlen)
 
 	new_sock = VTABLE_CALL(accept, sock, addr, addrlen);
 
-	(void)sock_obj_core_alloc_find(sock, new_sock, addr->sa_family, SOCK_STREAM);
+	if (addr) {
+		(void)sock_obj_core_alloc_find(sock, new_sock, addr->sa_family, SOCK_STREAM);
+	} else {
+		(void)sock_obj_core_alloc_find(sock, new_sock, 0, SOCK_STREAM);
```

Comment:
Sorry, we cannot do this as the address family would be wrong now. We need to use the proper protocol family here.
We could move the `sock_obj_core_alloc_find()` function to `zsock_accept_ctx()` at the end where we know the new fd. There `ctx->remote.sa_family` could be used as a family.

Reply:
Unfortunately I'm dealing with offloaded sockets so the vtable is different and thus `zsock_accept_ctx` would not get called. Also the suggested change would alter the standard signature of the `accept` function.

Would it be acceptable to simply keep the lines `687` and `688` of my suggested patch? This however would mean that in the (maybe rare) instances in which `addr` and `addrlen` are null, there won't be any registered statistic for these accept calls.

Reply:
> Would it be acceptable to simply keep the lines `687` and `688` of my suggested patch? This however would mean that in the (maybe rare) instances in which `addr` and `addrlen` are null, there won't be any registered statistic for these accept calls.

I suppose that is the only alternative then for time being.

Reply:
Maybe we should consider implementing `SO_DOMAIN` to obtain address family (not necessarily in this PR, could be a future improvement).

Reply:
> Maybe we should consider implementing `SO_DOMAIN` to obtain address family

Indeed, I can look into this and propose a PR.